### PR TITLE
Passing trailing arguments to test runners

### PR DESCRIPTION
**Pull request by rosic**
Pushing all the args after -- to ARGV before test unit file is required.

### CODE REVIEW

**Comment by rosic** [comment]
Hi, yesterday I was really suffering to run the whole file all the time, so I added a forwarding of all push args following the `--` to the child process.

It's the same issue as [#32](https://github.com/jstorimer/spin/pull/32) but with [#34](https://github.com/jstorimer/spin/issues/34) style. If you'd be interested, we can clean it up and generalize for both `Test::Unit` and `Rspec`.

**Comment by kirs** [comment]
@rosic yeah, it would be great to generalize it for both Test::Unit and RSpec. Also rebase in needed

**Comment by rosic** [comment]
Hey @kirs, haven't noticed that it has been refactored, because I was happily using this branch ever since :) Rebased, but I guess not completely adjusted to the current version.

File: lib/spin/cli.rb [hunk 1]
```diff
@@ -18,8 +18,8 @@ def run(argv)
             $LOAD_PATH.concat(dirs.split(File::PATH_SEPARATOR))
           end
 
-          opts.on("--rspec", "Force the selected test framework to RSpec") { options[:test_framework] = :testunit }
-          opts.on("--test-unit", "Force the selected test framework to Test::Unit") { options[:test_framework] = :rspec }
```
Comment (rosic) [inline lib/spin/cli.rb:22]: I found this confusing, was there an intention why `--rspec` was setting a framework to `testunit`? Didn't sound like RSpec is the one being forced (and vice versa).

**Comment by kirs** [comment]
@rosic great! I would be happy to merge it and release the new version of Spin then, but do you need some time to finalize the feature?

File: lib/spin/cli.rb (hunk 1 above)
Comment (kirs) [inline lib/spin/cli.rb:22]: I'm agree with you. Are the tests passing?

Comment (rosic) [inline lib/spin/cli.rb:22]: yep

**Comment by rosic** [comment]
@kirs All tests passed and worked for me in a project with RSpec as well as in another one with TestUnit. That depends on you or anyone else interested, if you want me to change something to fit better current design or it's good enough :)

**Comment by kirs** [comment]
Looks pretty good for me. Thank you for contributing spin! :+1:

**Comment by rosic** [comment]
Great, cheers!

**Comment by kirs** [comment]
https://travis-ci.org/jstorimer/spin/jobs/7553886 :worried:
Spin with simple setup RSpec can pass trailing arguments to the spec runner
     Failure/Error: served.should include 'Top 3 slowest examples'

**Comment by kirs** [comment]
It's failing :confused: could you plz check the spec?

**Comment by rosic** [comment]
Hey,
should be fixed here https://github.com/jstorimer/spin/pull/78

Oddly, the test passed for me before I've pushed the PR, but I have no clue
anymore what gems were loaded. Now with the updated gemspec and Gemfile it
was easy to reproduce and fix.

Sorry and thanks
Marek

2013/5/28 Kir Shatrov notifications@github.com

**Comment by kirs** [comment]
Merged. Thanks for the quick response!

**Comment by rosic** [comment]
No problem. But I see I broke 1.8.7 support with new hash syntax :/ Damn,
I'm really sloppy with spin.

2013/5/28 Kir Shatrov notifications@github.com

**Comment by kirs** [comment]
Don't worry, I'm gonna fix it in master soon :)

## 

Kir Shatrov
EU: +358 403640593, Russia: +7 (929) 631-46-35

**Comment by kirs** [comment]
Well, we still have some problems with 1.8.7 and trailing args: https://travis-ci.org/jstorimer/spin/jobs/7562586

**Comment by rosic** [comment]
I see, i'll try in the evening.

2013/5/28 Kir Shatrov notifications@github.com

**Pull request merged**